src/engine/ruleset/decoders/sonicwall/firewall-generated.yml

```diff
@@ -0,0 +1,166 @@
+        note=<~/ignore/ >"<~log.note>" npcs=<~rsa.db.index>
+    # id=ntutlabo sn=iusmodte time="2018-6-19 3:46:49" fw=10.108.84.24 pri=low c=iosamnis m=606 msg="volupt" n=rem src=10.113.100.237:3887:eth163 dst=10.251.248.228:6909 srcMac= 01:00:5e:8b:c1:b4 dstMac=01:00:5e:c3:ed:55proto=udp fw_action="deny"
+    # id=alo sn=eosquir time="2016-3-12 3:17:42" fw=10.149.203.46 pri=medium c=mwritten m=1369 msg="ctetur" n=uidolorsrc=10.150.156.22:6378:eth6183dst=10.227.15.1:410:eth1977srcMac=01:00:5e:84:66:6cdstMac=01:00:5e:f7:a9:ffproto=rdp/ommfw_action="allow"
+    - event.original: >-
+        id=<event.code> sn=<~rsa.misc.serial_number> time="<@timestamp/%Y-%m-%d %H:%M:%S>"(? )fw=(? )<host.ip>(? )pri=(? )<log.level>(? )c=(? )<~rsa.misc.category>(? )m=(? )<event.code>(? )msg=(? )"<~log.msg_action>"(? )n=(? )<~rsa.misc.ntype>(? )src=(? )<source.ip>:<source.port>(?:)<?observer.ingress.interface.name>(? )dst=(? )<destination.ip>:<destination.port>(?:)<?observer.egress.interface.name>(? )srcMac=(? )<source.mac>(? )dstMac=(? )<destination.mac>proto=<network.protocol>(? )fw_action=(? )"<~log.fw_action>"
+    # id=undeo sn=loremip time="2016-8-16 8:45:59" fw=10.134.0.141 pri=very-high c=uis m=1149 msg="idolore" n=onse fw_action="cancel"
+    - event.original: >-
+        <~log.fill> time="<@timestamp/%Y-%m-%d %H:%M:%S>" <~log.fill> m=<event.code> msg="<~log.msg_action>" <~log.fill> fw_action="<~log.fw_action>"
+
+normalize:
+  - map:
+      - event.kind: event
+      - event.module: sonicwall
+      - event.dataset: sonicwall.firewall
+      - fileset.name: firewall
+      - input.type: log
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time: $@timestamp
+      - rsa.network.dinterface: $observer.egress.interface.name
+      - rsa.network.sinterface: $observer.ingress.interface.name
+      - rsa.db.index: $~rsa.db.index
+      - tags: +a_append/forwarded/sonicwall.firewall
+      - rsa.identity.user_sid_dst: $~rsa.identity.user_sid_dst
+  - check:
+      - host.name: +ef_exists
+      - source.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$host.name/$source.address
+  - check:
+      - destination.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$destination.address
+      - rsa.network.host_dst: $destination.address
+  - check:
+      - source.ip: +ef_exists
+      - destination.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.ip/$destination.ip
+  - check:
+      - source.nat.ip: +ef_exists
+      - destination.nat.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.nat.ip/$destination.nat.ip
+  - check:
+      - ~log.start_message: +ef_not_exists
+    logpar:
+      - rsa.time.event_time: <~log.year>-<~log.mounth>-<~log.day>T<~log.fill>
```

Comment:
Please, don't use `logpar` on the `normalize` stage. You could make use of the `s_to_array` helper function in this case.